### PR TITLE
test (webapi): add gatling-based performance tests (part 1)

### LINKED ISSUE #4: Implement automated performance tests
- Test the performance and scalability of the Knora API server with the fake triplestore
- Test performance with different real triplestores
- Have the performance tests run automatically on the test server

### PR DESCRIPTION
**Pull request by subotic**
### Summary

- Scenario for testing the User Admin API:
  - [x] 1000 concurrent users accessing `/admin/user` (getting a list of all users)
  - [x] 25 concurrent users creating a user

For further finetuning, a better insight into the inner-workings during runtime is needed #2

### Issues

- relevant for #4 
- relevant for #133

### CODE REVIEW

**Comment by subotic** [comment]
@benjamingeer This is ready for review. I first need to implement #2 to have better insights for finetuning.

**Comment by benjamingeer** [comment]
Cool! I'll review it on Monday.

**Comment by subotic** [comment]
thanks :-)

File: webapi/src/test/scala/org/knora/webapi/e2e/ExampleE2ESimSpec.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+package org.knora.webapi.e2e
+
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+import org.knora.webapi.E2ESimSpec
+import org.knora.webapi.messages.store.triplestoremessages.RdfDataObject
+
+import scala.concurrent.duration._
+
+class ExampleE2ESimSpec extends E2ESimSpec {
```
Comment (benjamingeer) [inline webapi/src/test/scala/org/knora/webapi/e2e/ExampleE2ESimSpec.scala]: Class comment here, please.

File: webapi/src/test/scala/org/knora/webapi/e2e/admin/AdminMixE2ESimSpec.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+package org.knora.webapi.e2e.admin
+
+import io.gatling.core.Predef.{forAll, global, rampUsers, scenario, _}
+import io.gatling.http.Predef.{http, status}
+import org.knora.webapi.E2ESimSpec
+import org.knora.webapi.messages.store.triplestoremessages.RdfDataObject
+
+import scala.concurrent.duration._
+
+class AdminMixE2ESimSpec extends E2ESimSpec {
```
Comment (benjamingeer) [inline webapi/src/test/scala/org/knora/webapi/e2e/admin/AdminMixE2ESimSpec.scala]: Needs class comment.

File: webapi/src/test/scala/org/knora/webapi/e2e/admin/GroupsADME2ESimSpec.scala [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+import org.knora.webapi.E2ESimSpec
+import org.knora.webapi.messages.store.triplestoremessages.RdfDataObject
+
+import scala.concurrent.duration._
+
+
+class GroupsADME2ESimSpec extends E2ESimSpec {
```
Comment (benjamingeer) [inline webapi/src/test/scala/org/knora/webapi/e2e/admin/GroupsADME2ESimSpec.scala]: Needs class comment.

File: webapi/src/test/scala/org/knora/webapi/e2e/admin/ProjectsADME2ESimSpec.scala [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+import org.knora.webapi.E2ESimSpec
+import org.knora.webapi.messages.store.triplestoremessages.RdfDataObject
+
+import scala.concurrent.duration._
+
+
+class ProjectsADME2ESimSpec extends E2ESimSpec {
```
Comment (benjamingeer) [inline webapi/src/test/scala/org/knora/webapi/e2e/admin/ProjectsADME2ESimSpec.scala]: Needs class comment.

**Comment by subotic** [comment]
thanks. added class comments.

**Pull request merged**